### ISSUE TITLE
bayesian generate function can take an extremely long time (MNIST example)

### ISSUE THREAD

**Issue by qwertmullins** [keras-team/autokeras#635]
After completing training on 3 models, the program seems to hang

code:

(x_train, y_train), (x_test, y_test) = mnist.load_data()
x_train = x_train.reshape(x_train.shape + (1,))
x_test = x_test.reshape(x_test.shape + (1,))

clf = ImageClassifier(verbose=True, resume=True, path=r'mnist')
clf.fit(x_train, y_train, time_limit=12 * 60 * 60)
clf.final_fit(x_train, y_train, x_test, y_test, retrain=True)
y = clf.evaluate(x_test, y_test)
print(y)

output:

+----------------------------------------------+
|               Training model 3               |
+----------------------------------------------+
                                                                                                    
No loss decrease after 5 epochs.

Debugging shows the code is spinning in the while loop on line 350 of bayesian.py:

        while not pq.empty() and remaining_time > 0 and t > t_min:
            if isinstance(sync_message, Queue) and sync_message.qsize() != 0:
                break
            elem = pq.get()
            if self.metric.higher_better():
                temp_exp = min((elem.metric_value - opt_acq) / t, 1.0)
            else:
                temp_exp = min((opt_acq - elem.metric_value) / t, 1.0)
            ap = math.exp(temp_exp)
            if ap >= random.uniform(0, 1):
                for temp_graph in transform(elem.graph, skip_conn=self.skip_conn):
                    if contain(descriptors, temp_graph.extract_descriptor()):
                        continue

                    temp_acq_value = self.acq(temp_graph)
                    pq.put(elem_class(temp_acq_value, elem.father_id, temp_graph))
                    descriptors.append(temp_graph.extract_descriptor())
                    if self._accept_new_acq_value(opt_acq, temp_acq_value):
                        opt_acq = temp_acq_value
                        father_id = elem.father_id
                        target_graph = deepcopy(temp_graph)
            t *= alpha
            remaining_time = timeout - (time.time() - start_time)

pq has ~100 elements in it
ap was > 0.95 for the few iterations I checked, so the if statement seems to be true almost always
each iteration of the temp_graph for loop takes ~10 minutes

This results in the program hanging, seemingly indefinitely.  I'm not familiar with the intended flow at this point but something seems off.
